File: fibonacci-series/ruby/fibonacci.rb
def fibonacci(num)
  series = [0, 1]
  nth_value = 0

  for i in 1..num-1 do
    last = series[series.length-1]
    second_last = series[series.length-2]
    next_value = last + second_last

    series.concat([next_value])

    nth_value = next_value
  end

  nth_value
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: fibonacci-series/javascript/fibonacci.js
function fibonacci(num) {
  // input: num representing nth element of series (int; zero-indexed)
  // output: the value of that element of the series (int)

  let series = [0, 1];
  let nthValue = 0;

  for (let i = 1; i < num; i++) {
    let last = series[series.length-1];
    let secondLast = series[series.length-2]; 
    let nextValue = last + secondLast;

    series = [...series, nextValue];

    nthValue = nextValue;
  }

  return nthValue;
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));

  console.log("Expecting: some huge integer");
  console.log("=>", fibonacci(900));

  console.log("");

  console.log("Expecting: 89");
  console.log("=>", fibonacci(11));

  console.log("");

  console.log(`Expecting: ${55+89}`);
  console.log("=>", fibonacci(12));

  console.log("");
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution
